pycml/README.md

```diff
@@ -2,3 +2,44 @@
 
 These are not used in `fc`, but are here for reference while we switch from pycml
 to the new `cellmlmanip`/`weblab_cg` solution.
+
+
+## Some notes on how pycml processes the 'model interface' section
+
+Unless otherwise noted, bare method names are defined in `protocol.py`.
+Names with a `p:` prefix are defined in one of the classes in `processors.py`.
+
+See comments in the methods for details not mentioned here; this document focuses on the main flow and rationale.
+
+Many of the complexities in pycml arise from:
+- the XML object mapping representation for mathematics, which makes creating/changing maths complicated;
+- the need to put variables in components, and maintain sensible connections, including potentially through intermediate encapsulating components, ensuring unique naming;
+- arising from this, managing name references in mathematics (which can be local names, fully qualified names, or ontology terms) particularly when equations are moved or changed;
+- the need for units to be defined in a component and referenced by name from variables;
+- the internal state needed by code generation, for instance to classify variables (computed, state, etc) and topologically sort them. This state can shift as model modifications are made, and might not be fully known until all modifications have been done, so the state can sometimes be 'nonsense' at intermediate stages.
+
+Protocol parsing starts with `parse_protocol`, called by `apply_protocol_file`. It ensures that the different parts of the interface are processed in an order that respects the principle of least surprise for protocol authors: it tries to ensure their modifications do what they probably expect!
+
+Imports are treated first, included those arising implicitly from nested protocols. Where relevant, units definitions, interface definitions, etc. from imports are merged with the parent protocol's definitions.
```

Comment:
Does "arising implicitly" here mean "explicitly imported but in an imported protocol", or is there some other way that they can arise?

Reply:
Gory details are in the method, but in short, I think it's just model modifications (i.e. the model interface section) that matter from nested protocols.